### PR TITLE
Fixed uncapture in DropUnusedCaptures pattern.

### PR DESCRIPTION
**Pull request by KFAFSP**
Solves issue dropping unused captures.

This now mirrors the new implementation from DLNN, which is much cleaner anyways.

### CODE REVIEW

File: mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp [hunk 1]
```diff
@@ -826,33 +826,19 @@ struct DropUnusedCaptures : OpRewritePattern<FusedOp> {
...
       PatternRewriter &rewriter) const override {
-    // Compute a pruned mapping of captures values to actually used arguments.
-    BlockAndValueMapping prunedMapping, removedArgs;
-    for (unsigned idx = 0; idx < op.captures().size(); ++idx) {
-      auto arg = op.getCaptureArgs()[idx];
-      if (arg.getUses().empty()) {
-        removedArgs.map(arg, Value{});
-        continue;
+    // Find all unused arguments.
+    auto unused = llvm::to_vector(llvm::make_filter_range(
```
Comment (ljfitz) [inline mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp:830]: Thanks for solving this so quickly. Can you add a test that triggers the error with the original implementation?

**Review by gargaroff — APPROVED** [review]

File: mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp (hunk 1 above)
Comment (gargaroff) [reply mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp:830]: I should be able to create a test case based on the Makefile in my PR. I'll push it to this PR once I have it

Comment (gargaroff) [reply mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp:830]: @ljfitz Unfortunately mlir-reduce wasn't able to reduce the module, as it itself crashes randomly. Should we merge the PR without a regression test or should we use the unmodified module as a test instead?

Comment (ljfitz) [reply mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp:830]: Thanks for trying to reproduce the issue in a smaller testcase. Given a replacement is already in preparation, we can accept the gap in testing for now.

**Pull request merged**